tests/Unit/CollectionEqualsTest.php

```diff
@@ -42,6 +43,59 @@ public static function sameCollection(): array
         ];
     }
 
+    public static function sameUnorderedCollection(): array
+    {
+        Model::unguard();
+
+        $dataJohn = new SimpleData('John', 25);
+        $dataJack = new SimpleData('Jack', 30);
+
+        return [
+            [new Collection([1, 2, 3]), new Collection([2, 1, 3])],
+            [[1, 2, 3], new Collection([1, 3, 2])],
+            [new Collection([1 => 1, 0 => 2, 3 => 3]), new Collection([0 => 2, 1 => 1, 3 => 3])],
+            [[1 => 1, 0 => 2, 3 => 3], new Collection([1 => 1, 3 => 3, 0 => 2])],
+            [new Collection(['a' => 1, 'b' => 2, 'c' => 3]), new Collection(['a' => 1, 'c' => 3, 'b' => 2])],
+            [['c' => 3, 'a' => 1, 'b' => 2], new Collection(['a' => 1, 'b' => 2, 'c' => 3])],
+            [
+                new Collection([new User(['id' => 1, 'name' => 'John']), new User(['id' => 2, 'name' => 'Jack'])]),
+                new Collection([new User(['id' => 2, 'name' => 'Jack']), new User(['id' => 1, 'name' => 'Peter'])]),
+            ],
+            [
+                new Collection([
+                    'John' => new User(['id' => 1, 'name' => 'John']),
+                    'Jack' => new User(['id' => 2, 'name' => 'Jack']),
+                ]),
+                new Collection([
+                    'Jack' => new User(['id' => 2, 'name' => 'Jack']),
+                    'John' => new User(['id' => 1, 'name' => 'John']),
+                ]),
+            ],
+            [
+                new Collection([
+                    CarbonImmutable::create(2022, 2, 11),
+                    CarbonImmutable::create(2022, 5, 11),
+                ]),
+                new Collection([
+                    CarbonImmutable::createFromFormat('!Y-m-d', '2022-05-11'),
+                    CarbonImmutable::createFromFormat('!Y-m-d', '2022-02-11'),
+                ]),
+            ],
+            [
+                new Collection([new Collection([1, 2, 3]), new Collection([2, 3, 1])]),
+                new Collection([new Collection([2, 3, 1]), new Collection([1, 2, 3])]),
+            ],
+            [
+                new Collection([[1, 2, 3], [2, 1, 3]]),
+                new Collection([[2, 1, 3], [1, 2, 3]]),
+            ],
+            [
+                [$dataJohn, $dataJack],
```

Comment:
Inline data creation to show that comparison is not made by object reference but by data properties